tests/utils/inputparams/testparams.go

```diff
@@ -96,52 +101,56 @@ func GetEndPoint2() string {
 
 // GetSwarmManager1 returns swarm manager node IP from the configured swarm cluster
 func GetSwarmManager1() string {
-	return os.Getenv("MANAGER1")
+	manager1 := os.Getenv("MANAGER1")
+	return manager1
 }
 
 // GetSwarmWorker1 returns 1st swarm worker node IP from the configured swarm cluster
 func GetSwarmWorker1() string {
-	return os.Getenv("WORKER1")
+	worker1 := os.Getenv("WORKER1")
+	return worker1
 }
 
 // GetSwarmWorker2 returns 2nd swarm worker node IP from the configured swarm cluster
 func GetSwarmWorker2() string {
-	return os.Getenv("WORKER2")
+	worker2 := os.Getenv("WORKER2")
+	return worker2
 }
 
 // GetSwarmNodes returns all nodes in the configured swarm cluster
 func GetSwarmNodes() []string {
 	return []string{GetSwarmManager1(), GetSwarmWorker1(), GetSwarmWorker2()}
 }
 
-// GetDockerHostIP - returns ip of the VM where vm can be first vm (VM1) or second vm (VM2)
-func GetDockerHostIP(vm string) string {
-	return os.Getenv(vm)
-}
-
 // GetEsxIP returns the ip of the esx
 func GetEsxIP() string {
-	return os.Getenv("ESX")
+	esxIP := os.Getenv("ESX")
+	if esxIP == "" {
+		log.Fatal("ESX host not found. Stopping the test run.")
+	}
+	return esxIP
 }
 
 // GetTestConfig - returns the configuration of IPs for the
 // ESX host, docker hosts and the datastores on the host
-func GetTestConfig() *TestConfig {
-	var config *TestConfig
-
+func getInstance() *TestConfig {
+	noVMName := "no such VM"
 	config = new(TestConfig)
-	config.EsxHost = os.Getenv("ESX")
+	config.EsxHost = GetEsxIP()
 	config.DockerHosts = append(config.DockerHosts, os.Getenv("VM1"))
 	config.DockerHosts = append(config.DockerHosts, os.Getenv("VM2"))
+	if config.DockerHosts[0] == "" && config.DockerHosts[1] == "" {
+		log.Fatal("No docker hosts found. At least one host is needed to run tests.")
+	}
 	config.DockerHostNames = append(config.DockerHostNames, esx.RetrieveVMNameFromIP(config.DockerHosts[0]))
 	config.DockerHostNames = append(config.DockerHostNames, esx.RetrieveVMNameFromIP(config.DockerHosts[1]))
+	if config.DockerHostNames[0] == noVMName && config.DockerHostNames[1] == noVMName {
```

Comment:
The AND logic here should be OR, since we want to check both. Please fix the logs message below appropriately.